Sources/Meta/Extensions/String+NonArray.swift

```diff
@@ -0,0 +1,18 @@
+extension String {
+    var nonArray: String {
+        substring(pattern: "(?<=\\[).*(?=\\])")
+            .substring(pattern: "(?<=ArrayK<).*(?=>)")
+            .substring(pattern: "(?<=Array<).*(?=>)")
+            .substring(pattern: "(?<=NEA<).*(?=>)")
+            .substring(pattern: "(?<=NonEmptyArray<).*(?=>)")
+            .trimmingCharacters
+    }
+    
+    var isArrayType: Bool {
+        contains("[") ||
+        contains("Array<") ||
+        contains("ArrayK<") ||
+        contains("NEA<") ||
+        contains("NonEmptyArray<")
+    }
```

Comment:
We need to revisit this logic as `Foo<[Int]>` has `isArrayType == true`, but that is not correct. Also, we need to make sure we do not strip 2 layers of arrays when extracting things: `[[Int]]` needs to yield `[Int]`, not just `Int`.